src/ArcgisAccount/accountManagerUtils.js

```diff
@@ -0,0 +1,236 @@
+  }
+};
+
+//** Login */
+export const loginOAuth2 = async ({
+  clientId,
+  redirectUri,
+  portalUrl,
+  popup,
+  name,
+  setAccountManagerState
+}) => {
+  beginOAuthSignIn({
+    clientId,
+    redirectUri,
+    portalUrl,
+    popup,
+    name,
+    setAccountManagerState
+  });
+};
+
+//** Logout */
+export const logoutOAuth2 = ({ url, clientId, token }) => {
+  //** Formal Logout requires esri_auth cookie */
+  //AGO
+  //return `${portalUrl.origin}/home/pages/Account/manage_accounts.html#redirect_uri=${encodedRedirectUrl}&client_id=${params.clientId}&signout=true`
+  //Enterprise
+  //return `${portalUrl}/oauth2/signout?client_id=${params.clientId}&redirect_uri=${encodedRedirectUrl}`;
+
+  //** Revoke Token (OAuth2): https://developers.arcgis.com/rest/users-groups-and-items/revoke-token.htm */
+  //Build logout request
+  if (!clientId || !token) {
+    return { success: false, error: 'Empty client_id / token' };
+  }
+  const revokeTokenEndpoint = `${url}/oauth2/revokeToken`;
+  const data = {
+    client_id: clientId,
+    auth_token: token
+  };
+
+  try {
+    request(revokeTokenEndpoint, {
+      params: data,
+      httpMethod: 'POST'
+    }).then(res => {
+      return { success: res.success };
+    });
+  } catch (err) {
+    return { success: false, error: err };
+  }
+};
+
+//** Get Org Thumbnail */
+export const getOrgThumbnail = ({
+  session: { portal },
+  portal: { thumbnail, name },
+  token
+}) => {
+  //org thumbnail (https://developers.arcgis.com/rest/users-groups-and-items/portal-self.htm)
+  let orgThumbnail = {
+    url: undefined,
+    letters: undefined
+  };
+  orgThumbnail.url =
+    thumbnail && portal && token
+      ? `${portal}/portals/self/resources/${thumbnail}?token=${token}`
+      : undefined;
+  orgThumbnail.letters = name ? name[0].toUpperCase() : 'A';
+  return orgThumbnail;
+};
+
+//** Get User Thumbnail */
+export const getUserThumbnail = ({
+  session: { portal, username },
+  user: { thumbnail, firstName, lastName, fullName },
+  token
+}) => {
+  //user thumbnail (https://developers.arcgis.com/rest/users-groups-and-items/user.htm)
+  let userThumbnail = {
+    url: undefined,
+    letters: undefined
+  };
+  userThumbnail.url =
+    thumbnail && portal
+      ? `${portal}/community/users/${username}/info/${thumbnail}?token=${token}`
+      : undefined;
+
+  userThumbnail.letters =
+    firstName && lastName
+      ? `${firstName[0] + lastName[0]}`.toUpperCase()
+      : fullName
+      ? fullName[0].toUpperCase()
+      : undefined;
+  return userThumbnail;
+};
+
+const createAccountObject = async ({ dSession, portal, clientId }) => {
+  dSession.clientId = dSession.clientId ? dSession.clientId : clientId;
+  const token = dSession.token;
+  const user = await dSession.getUser();
+  const dPortal = await getPortal({ portalUrl: portal, session: dSession });
+
+  return {
+    session: dSession.serialize(),
+    user: user,
+    portal: JSON.stringify(dPortal),
+    token: token
+  };
```

Comment:
We could use Object Initializer shorthand here:

```suggestion
  return {
    session: dSession.serialize(),
    portal: JSON.stringify(dPortal),
    user,
    token
  };
```